test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -1097,6 +1174,35 @@ public Builder screenCapture() {
             return this;
         }
 
+        /**
+         * Adds a log area below the "Pass", "Fail" buttons.
+         * <p> The log area can be controlled by {@link #log(String)},
+         * {@link #logClear()} and {@link #logSet(String)}.
+         *
+         * @return this builder
+         */
+        public Builder logArea() {
+            this.addLogArea = true;
+            return this;
+        }
+
+        /**
+         * Adds a log area below the "Pass", "Fail" buttons.
+         * <p> The log area can be controlled by {@link #log(String)},
+         * {@link #logClear()} and {@link #logSet(String)}.
+         *
+         * <p> The number of columns is taken from the number of
+         * columns in the instructional JTextArea.
```

Comment:
This may be not what we want…

I'd like to get rid of specifying rows and columns for instructions, see [JDK-8328163](https://bugs.openjdk.org/browse/JDK-8328163). I can't remember why we decided to specify those explicitly.

On the other hand, if adjustment is required, `.columns` will do the job for both cases, so adding another parameter is likely an overkill.

Reply:
It does the job well for now, so we can get to that later when the actual automatic instruction size is implemented.